### PR TITLE
Commander does not allow upgrading using a custom build - Closes #3901 #3898 #3911

### LINKED ISSUE #3898: Commander should only download snapshots for mainnet and testnet
### Expected behavior
`lisk core:install --network=devnet lisk-devnet` does not try to download a blockchain snapshot since there is no such snapshot for `devnet`.

### Actual behavior
`lisk core:install --network=devnet lisk-devnet` fails trying to download a blockchain snapshot which does not exist.

### Steps to reproduce
`lisk core:install --network=devnet lisk-devnet`

### Which version(s) does this affect? (Environment, OS, etc...)
`2.2.0`

### LINKED ISSUE #3901: Commander does not allow upgrading using a custom build
### Expected behavior
```
lisk core:install --network devnet --release-url=http://localhost/lisk-2.0.0-rc.0-Linux-x86_64.tar.gz --no-snapshot lisk-devnet
lisk core:upgrade --release-url=http://localhost/lisk-2.0.0-rc.1-Linux-x86_64.tar.gz lisk-devnet
```
File is downloaded from `http://localhost/lisk-2.0.0-rc.1-Linux-x86_64.tar.gz`

### Actual behavior
```
./lisk/bin/lisk core:upgrade --release-url=http://localhost/lisk-2.0.0-rc.1-Linux-x86_64.tar.gz lisk-devnet
  ✖ Validate Version Input
    → Upgrade version: 2.0.0-rc.0- doesn't exists in https://downloads.lisk.io/lisk/devnet
    Download Lisk Core: 2.0.0-rc.0- for upgrade
    Stop, Backup and Install Lisk Core
    Upgrade Lisk Core lisk-devnet instance from: 2.0.0-rc.0 to: 2.0.0-rc.0-1
    Unregister and register Lisk Core
    Start Lisk Core: 2.0.0-rc.0-
 ›   Error: Upgrade version: 2.0.0-rc.0-1 doesn't exists in https://downloads.lisk.io/lisk/devnet
```

The file is not being downloaded from the URL passed as a parameter.
Additionally, trying to upgrade from `2.0.0-rc.0` to `2.0.0-rc.0` leads to the above error and also the version being wrongly reported as `2.0.0-rc.0-`.

### Steps to reproduce
Try using the `--release-url` option of the `core:upgrade` command.

### Which version(s) does this affect? (Environment, OS, etc...)
`2.2.0`

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?
- Upgrading devnet using custom url was broken
- Commander should not download snapshot for devnet/alphanet/betanet
### How did I fix it?
- Allowing custom url for upgrade 
- Restricting snapshot download for devnet/alphanet/betanet 
### How to test it?
```
lisk core:install --network devnet --release-url=http://localhost/lisk-2.0.0-rc.0-Linux-x86_64.tar.gz --no-snapshot lisk-devnet
lisk core:upgrade --release-url=http://localhost/lisk-2.0.0-rc.1-Linux-x86_64.tar.gz lisk-devnet
```
### Review checklist

* The PR resolves #3898 #3901 #3911 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by fchavant** [comment]
"Upgrading" from `2.0.0-rc.0` to `2.0.0-rc.0` is still broken (`2.0.0-rc.0-`):

``` 
lisk core:upgrade --release-url=http://localhost/lisk-2.0.0-rc.0-Linux-x86_64.tar.gz lisk-devnet
[...]
lisk core:upgrade --release-url=https://localhost/lisk-2.0.0-rc.0-Linux-x86_64.tar.gz lisk-devnet
  ✔ Validate Version Input
  ✔ Download Lisk Core: 2.0.0-rc.0- for upgrade
  ✔ Stop, Backup and Install Lisk Core
  ✔ Upgrade Lisk Core lisk-devnet instance from: 2.0.0-rc.0 to: 2.0.0-rc.0-
  ✔ Unregister and register Lisk Core
  ✔ Start Lisk Core: 2.0.0-rc.0-
```

@ManuGowda I understand commander is not supposed to let one "upgrade" from a given version to the exact same version.

**Comment by ManuGowda** [comment]
> Looks good now although I cannot find any backup after `core:upgrade`:
> 
> ```
> ~> ls -l ~/.lisk/instances/backup/
> total 0
> ```

ok I will look into it 👍

**Comment by ManuGowda** [comment]
@fchavant `backup` issue is fixed, here is the build for testing https://jenkins.lisk.io/job/lisk-qa/job/lisk-commander-build-tag/32/

**Comment by fchavant** [comment]
@ManuGowda upgrade/backup now works. My only complaints is that the backups get created under `~/.lisk/backup/` instead of `~/.lisk/backups/` (plural, like `~/.lisk/instances/`)

**Comment by ManuGowda** [comment]
@fchavant renamed from `backup` to `backups`

File: commander/src/utils/core/commons.ts [hunk 1]
```diff
@@ -131,13 +139,13 @@ export const getVersionToInstall = async (
...
-	);
-	if (stderr) {
-		throw new Error(stderr);
+export const backupLisk = (installDir: string, instanceName: string): void => {
+	try {
+		const backupPath = `${defaultBackupPath}/${instanceName}`;
+		fsExtra.removeSync(backupPath);
+		fsExtra.moveSync(installDir, backupPath);
+	} catch (error) {
+		throw new Error(error);
```
Comment (shuse2) [inline commander/src/utils/core/commons.ts]: what is this error type?
I think no need to catch and re-throw in this case

**Comment by fchavant** [comment]
@ManuGowda does not work as of [build #33](https://jenkins.lisk.io/job/lisk-qa/job/lisk-commander-build-tag/33/): backup directory remains after `core:uninstall`.

**Pull request merged**